### PR TITLE
CO Renaming (Pt. 1)

### PR DESCRIPTION
**Pull request by Holzhaus**
*This PR is part of the ongoing effort to remove offensive language from the Mixxx codebase (#11931).*

Here is the first bunch of renamings discussed in https://github.com/mixxxdj/mixxx/pull/11960#issuecomment-1714591738. I removed the `num_mics_configured` CO since it's used nowhere outside the `EngineMixer` class (not even mappings or skins) and can be replaced with a simple member variable.

| Old Name                                                | New Name                                              | Alias?
| ------------------------------------------------------- | ----------------------------------------------------- | ----------
| `[Master],guiTick50ms`                                  | `[App],gui_tick_50ms_period_s`                             | :heavy_check_mark: (although undocumented)
| `[Master],guiTickTime`                                  | `[App],gui_tick_full_period_s`                                 | :heavy_check_mark: (although undocumented)
| `[Master],indicator_250millis`                          | `[App],indicator_250ms`                           | :heavy_check_mark:
| `[Master],indicator_500millis`                          | `[App],indicator_500ms`                           | :heavy_check_mark:
| `[Master],keylock_engine`                               | `[App],keylock_engine`                                | :x:
| `[Master],num_auxiliaries`                              | `[App],num_auxiliaries`                               | :heavy_check_mark: (although undocumented)
| `[Master],num_decks`                                    | `[App],num_decks`                                     | :heavy_check_mark:
| `[Master],num_microphones`                              | `[App],num_microphones`                               | :heavy_check_mark: (although undocumented)
| `[Master],num_mics_configured`                          | **Removed**                                           | n/A
| `[Master],num_preview_decks`                            | `[App],num_preview_decks`                             | :heavy_check_mark:
| `[Master],num_samplers`                                 | `[App],num_samplers`                                  | :heavy_check_mark:
| `[Master],samplerate`                                   | `[App],samplerate`                                    | :heavy_check_mark:

### CODE REVIEW

**Comment by Swiftb0y** [comment]
While were at it, I'd also like to propose to retire the `millis` suffix.  We had that as a `suffix` for constants and member variables for some time, but this will get obsolete with `<chrono>` types and literals fast. So I'd propose to stick with the `ms` actually since that mirrors `std::chrono_literals`. Wdyt?

File: src/engine/enginebuffer.cpp
Comment (Swiftb0y) [inline src/engine/enginebuffer.cpp:263]: unrelated to the change, but we might want to consider switching `ConfigKey` from `QString` to `QLatin1String` as those can be significantly faster apparently and I don't think we need/want Unicode support in COs, do we?

File: src/engine/enginemixer.h
Comment (Swiftb0y) [inline src/engine/enginemixer.h:339]: Did you check that this is needed due to concurrent access or was this just a safe guess?

File: src/test/autodjprocessor_test.cpp
Comment (JoergAtGithub) [inline src/test/autodjprocessor_test.cpp:133]: Please only add a test for the new CO name and leave the existing one as it is

**Review by JoergAtGithub — CHANGES_REQUESTED** [review]

File: src/engine/enginemixer.h
Comment (Holzhaus) [reply src/engine/enginemixer.h:339]: `EngineMixer::process` reads the CO in the engine thread and I think the `onInputConnected`/`onInputDisconnected` methods that modify the value are likely triggered from the GUI thread (although I didn't check).

File: src/engine/enginebuffer.cpp
Comment (Holzhaus) [reply src/engine/enginebuffer.cpp:263]: This makes sense, but I'd rather not do this here, because it would mean we need to change *a lot* of COs. This can go into `main` IMHO.

**Comment by Holzhaus** [comment]
> While were at it, I'd also like to propose to retire the `millis` suffix.  We had that as a `suffix` for constants and member variables for some time, but this will get obsolete with `<chrono>` types and literals fast. So I'd propose to stick with the `ms` actually since that mirrors `std::chrono_literals`. Wdyt?

Okay for me. By the way, should I also rename `gui_tick_time` to `gui_tick_s` then?

File: src/engine/enginebuffer.cpp
Comment (Swiftb0y) [reply src/engine/enginebuffer.cpp:263]: Yes, I agree. Just wanted to raise the point so I don't forget about it.

**Comment by Swiftb0y** [comment]
> Okay for me. By the way, should I also rename gui_tick_time to gui_tick_s then? I think it's in seconds if I remember correctly.

Sure. though I didn't check either, never heard of that CO before tbh...

File: src/test/autodjprocessor_test.cpp
Comment (Holzhaus) [reply src/test/autodjprocessor_test.cpp:133]: The plan was to remove the word from the codebase as much as possible which includes tests. Also, these tests are supposed to check if the AutoDJ processor works correctly, not if old CO names still work. I agree that it makes sense to test this, too, but that should be a different test IMHO. 

I was planning to add a new test that checks if the aliases exist. This would ideally go into `coreservicestest.cpp` since most other tests do not create all COs. Unfortunately this test is broken at the moment. I'm currently looking into it.

Comment (Swiftb0y) [reply src/test/autodjprocessor_test.cpp:133]: this would / could result in warnings raised by tests though if we decided to implement the warnings on deprecated CO access.

Comment (JoergAtGithub) [reply src/test/autodjprocessor_test.cpp:133]: We need to ensure the backward compatibility of our legacy API. I'm open for any test implementation, that ensures this.

**Comment by daschuer** [comment]
'guiTickTime' and 'guiTick50ms' have are both in seconds in terms of std::chrono. 
The difference is that the guiTickTime is advanced in refresh rate steps and guiTick50ms is advanced in 50ms steps. The diffrence is the period in terms of chono. 

So we may think of
'gui_full_period_tick_s'
'gui_50ms_period_tick_s'

**Comment by daschuer** [comment]
Derived form that are
'indicator_250millis'
'indicator_500millis'

They just cycle 0.0 ad 1.0. 
Rename as well?

File: src/widget/wtracktableview.cpp
Comment (Swiftb0y) [inline src/widget/wtracktableview.cpp:64]: this dependency is... interesting... worrisome? 
Anyways, obsolete with QML.

File: res/controllers/Traktor-Kontrol-S4-MK3.js
Comment (Swiftb0y) [inline res/controllers/Traktor-Kontrol-S4-MK3.js:2888]: we should probably track this in an issue on the QML project board, otherwise this will silently break under QML. putting this on an `engine.beginTimer` instead should be sufficient.

**Review by Swiftb0y — APPROVED** [review]
LGTM

**Review by Swiftb0y — APPROVED** [review]
LGTM. last chance for anyone to object.

**Comment by Holzhaus** [comment]
I was planning to add tests for the legacy aliases as @JoergAtGithub requested, but at the `CoreServicesTest` does not work on CI. I'll try to find another way.

**Comment by JoergAtGithub** [comment]
> I was planning to add tests for the legacy aliases as @JoergAtGithub requested, but at the `CoreServicesTest` (which would be the ideal location for such a test because it creates all COs) does not work on CI. I'll try to find another way.

I appreciate your effort! I think it's safer to have this test case in place before merging this. Otherwise this is LGTM!

File: src/test/controlobjectaliastest.cpp
Comment (Holzhaus) [inline src/test/controlobjectaliastest.cpp:24]: I think this needs to be removed when merging into main, right?

**Comment by Holzhaus** [comment]
> > I was planning to add tests for the legacy aliases as @JoergAtGithub requested, but at the `CoreServicesTest` (which would be the ideal location for such a test because it creates all COs) does not work on CI. I'll try to find another way.
> 
> I appreciate your effort! I think it's safer to have this test case in place before merging this. Otherwise this is LGTM!

Tests are in place now.

**Review by JoergAtGithub — APPROVED** [review]
LGTM! Waiting for CI

**Comment by JoergAtGithub** [comment]
Thank you!

**Comment by Swiftb0y** [comment]
Thank you as well.

**Comment by Holzhaus** [comment]
https://github.com/mixxxdj/manual/pull/579

**Pull request merged**
